test/cpp/test_status_common.h

```diff
@@ -14,80 +32,78 @@ namespace torch_xla {
 
 // Enum to control whether C++ error context is shown in status messages
 enum class CppErrorContextMode {
-  SHOW,
-  HIDE,
+  kShow,
+  kHide,
 };
 
 // Converts CppErrorContextMode enum to string for test parameter naming
 inline const char* const ToString(CppErrorContextMode mode) {
   switch (mode) {
-    case CppErrorContextMode::SHOW:
+    case CppErrorContextMode::kShow:
       return "ShowCppErrorContext";
-    case CppErrorContextMode::HIDE:
+    case CppErrorContextMode::kHide:
       return "DontShowCppErrorContext";
   }
 }
 
 // Base test class for parameterized status tests with C++ error context control
 class StatusTest : public testing::TestWithParam<CppErrorContextMode> {
-  void SetUp() override {
+ public:
+  StatusTest() {
     const char* const value = IsShowCppErrorContextMode() ? "true" : "false";
     setenv(runtime::env::kEnvShowCppErrorContext, value, /* replace= */ 1);
   }
 
- public:
+ protected:
   bool IsShowCppErrorContextMode() {
-    return GetParam() == CppErrorContextMode::SHOW;
+    return GetParam() == CppErrorContextMode::kShow;
   }
 };
 
 // Macro to instantiate parameterized status tests with specific mode
-#define INSTANTIATE_TEST_SUITE_WITH_MODE(mode)                            \
-  using torch_xla::CppErrorContextMode;                                   \
-  using torch_xla::StatusTest;                                            \
-  INSTANTIATE_TEST_SUITE_P(                                               \
-      StatusTest, StatusTest, testing::Values(CppErrorContextMode::mode), \
-      [](const testing::TestParamInfo<CppErrorContextMode>& info) {       \
-        return ToString(info.param);                                      \
+//
+// Note that this macro assumes that both `torch_xla::StatusTest` and
+// `torch_xla::CppErrorContextMode` are accessible from the context this macro
+// is being called.
+#define INSTANTIATE_WITH_CPP_ERROR_CONTEXT_MODE(suite, test, mode)  \
+  INSTANTIATE_TEST_SUITE_P(                                         \
+      suite, test, testing::Values(CppErrorContextMode::mode),      \
```

Comment:
Style: in a macro body, always fully qualify the symbols (use absolute namespaces like `::testing::`), so that we don't have to worry about changing the macro meaning if it's used in the wrong namespace.